### PR TITLE
plug.sh: reduce subshells / exec's

### PR DESCRIPTION
**Pull request by mralusw**
I've replaced a few `expr $HAYSTACK : .*$NEEDLE.*` with the equivalent `case $HAYSTACK in (*$NEEDLE*)`.

The `git rev-parse ARG1 ARG2..` usage (instead of multiple `git rev-parse ARG` calls) works on at least git 2.20.1 (I've just tested in a clean `debootstrap`'ed debian oldstable).

Reading from multiple `$()`'s, grouped in a heredoc is POSIX.

### CODE REVIEW

File: rc/plug.sh [hunk 1]
```diff
@@ -20,20 +20,21 @@ plug () {
...
-    fi
+        ;;
+    esac
 
     while [ $# -gt 0 ]; do
         case $1 in
             (branch|tag|commit) checkout_type=$1; shift; checkout="$1" ;;
             (noload) noload=1 ;;
-            (load-path) shift; path_to_plugin=$(eval echo "$1") ;;
+            (load-path) shift; eval "path_to_plugin=$1" ;;
```
Comment (andreyorst) [inline rc/plug.sh:37]: I think this might break if env variable is used in the path.

Comment (andreyorst) [reply rc/plug.sh:37]: oh, I think originaly I've added it to resolve tilda `eval echo "~/"` will print fully qualified path

Comment (mralusw) [reply rc/plug.sh:37]: I figured you wanted to resolve vars / tildas, which is why I kept the eval. But the two are equivalent, try it out:
```
(set -- '~/a'; path_to_plugin=$(eval echo "$1"); echo "$1 -> $path_to_plugin")
(set -- '~/a'; eval "path_to_plugin=$1"; echo "$1 -> $path_to_plugin")
```

I'm just removing a level of echo/capture indirection.

This is a shell gotcha... Note that `a=*; echo "$a"` prints `*` (no globbing), but `a=~; echo "$a"` and `a=$HOME; echo "$a"` resolve vars AND tildas.

Comment (mralusw) [reply rc/plug.sh:37]: ... of course, both forms resolve not just the initial tilda / `$HOME`, but also any `$component` / `${expansion}` that appears in the path , whether the user meant or not. But it seems a good tradeoff.

Comment (mralusw) [reply rc/plug.sh:37]: ... and, of course, also any `$(rm -rf ~)` appearing the path string (sorry for the comment spam). Given that users are installing unreviewed code anyway, it doesn't make much difference.

**Comment by andreyorst** [comment]
>
> Given that users are installing unreviewed code anyway, it doesn't make
> much difference.
>

It's user entered code, so I guess it's ok...

 ...hopefully no one will do something like plug "foo" load-parh %!echo
'test... test... test...' | perl -e '$??s:;s:s;;$?::s;;=]=>%-{<-|}<&|`{;;y;
***@***.***`{-};`-{/" -;;s;;$_;see'!

(seriously, don't do that, you will lose everything)

**Comment by andreyorst** [comment]
Thanks!

**Pull request merged**